lib/models/room.js

```diff
@@ -185,6 +191,18 @@ Room.prototype.getAvatarUrl = function(baseUrl, width, height, resizeMethod,
  };
 
  /**
+  * Get the implicit room name (i.e. what a given user would see if the
+  * room had no m.room.name)
+  * @param {string} userId The userId from whose perspective we want
+  * to calculate the implicit name
+  * @return {string} The implicit room name
+  */
+ Room.prototype.getImplicitRoomName = function(userId) {
```

Comment:
What are you planning on using this for? It seems to be conflating 2 ideas and has a rather vague function name, so I'm immediately skeptical. The 2 ideas being conflated are:
 - Getting the room name assuming there is no `m.room.name` event.
 - Getting the room name from the perspective of another user ID.

I don't know why you'd want to do either of these things, hence asking for more information on the intended usage.